src/compiler.ts

```diff
@@ -7165,6 +7213,25 @@ export class Compiler extends DiagnosticEmitter {
         let isUnchecked = this.currentFlow.is(FlowFlags.UNCHECKED_CONTEXT);
         let indexedGet = classReference.lookupOverload(OperatorKind.INDEXED_GET, isUnchecked);
         if (indexedGet) {
+          if (indexedGet.is(CommonFlags.STATIC)) {
+            this.error(
+              DiagnosticCode._0_modifier_cannot_be_used_here,
+              indexedGet.identifierNode.range, CommonNames.static_
+            );
+            return module.unreachable();
```

Comment:
Another `this.currentType`.

Reply:
Basically this missing in all other branches with early error return. I just followed the same behavior. Is it really important?

Reply:
When we have the info, updating makes sense since otherwise `this.currentType` is whatever it was before, and follow-up diagnostics might print types that don't really make any sense in context. I think the only place where it can be safely omitted is if we are sure that `currentType` does not need an update.